src/index.js

```diff
@@ -150,31 +160,31 @@ module.exports = {
 
   sourceDirsFor(strategy) {
     return this.optionFor(strategy, 'sourceDirs')
-      .filter((sourceDir) => fs.existsSync(sourceDir));
+      .filter((sourceDir) => typeof sourceDir !== 'string' || fs.existsSync(sourceDir));
```

Comment:
This was needed because we may sometimes discover that the app's default `public` tree is already some kind of broccoli output tree and not a plain string.